### PR TITLE
Propagate environment variables to container fixes #1266

### LINKED ISSUE #1266: ToolConfigs to auto-propagate env variables to singularity
Our ToolConfig model allows the admin to configure environment variables settings for the tool, e.g. `ABC=xyz`. These setting are transformed into a set of `export ABC=xyz` statements in the job script.

When a tool is run within Singularity, these environment variables will be propagated by default into the container, except if the admin configures the `--cleanenv` (or `-c`) option when singularity is invoked (in the ToolConfig attribute `container_exec_args` which shows up as 'Misc Singularity options in the page). In that case the admin must either

1. add explicitly another `--env ABC=xyz` (or better `--env ABC=$ABC`) in the Singularity options
2. change the name of the environment variables in the ToolConfig env table so they get prefixed with "SINGULARITYENV_" (e.g.  "SINGULARITYENV_ABC")

Option 1 means redundancy in defining the environment variables, and option 2 means that the tool config's environment table is now 'aware' of the containerized situation, which can lead to inconsistency when configuring the exact same tool without a container.

In order to make the life of the admin easier, I suggest that whenever a tool is launched through singularity, the set of export statements created by the **ClusterTask** object be replaced (or extended) with assignments of the form `export SINGULARITYENV_ABC=xyz`.

### PR DESCRIPTION
**Pull request by MontrealSergiy**

### CODE REVIEW

File: BrainPortal/cbrain_plugins/cbrain-plugins-base/boutiques_descriptors/singularity_env_demo.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            "list":          false,
+            "path-template": "old*"
+        },
+        {
+            "name":          "new file",
+            "id":            "s2",
+            "description":   "The output of the 'du' command in human units",
+            "optional": true,
+            "list":          false,
+            "path-template": "new"
```
Comment (natacha-beck) [inline BrainPortal/cbrain_plugins/cbrain-plugins-base/boutiques_descriptors/singularity_env_demo.json:35]: Should be `new*` no ?

**Comment by MontrealSergiy** [comment]
@natacha-beck ok the description is  contrived. It is test/demo descriptor for the new module. May be better delete it or place in a separate  elsewhere?

File: BrainPortal/cbrain_plugins/cbrain-plugins-base/boutiques_descriptors/singularity_env_demo.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{
+    "name":           "ENVPrefix",
+    "tool-version":   "1.0.0",
+    "schema-version": "0.5",
+    "author":         "Serge",
+    "description":    "A demo CBRAIN task that adds SINGULARITYENV_ prefix to config environment variable names ",
+
+    "command-line": "touch old_{$MYVARIABLE}; touch new_{$SINGULARITYENV_MYVARIABLE} ; du -a -H -h [CBRAIN_INPUT] > out",
```
[the remark this answers is not in the document]
Comment (prioux) [reply BrainPortal/cbrain_plugins/cbrain-plugins-base/boutiques_descriptors/singularity_env_demo.json:8]: That is bad bash syntax.

Substitution of variables in bash is either plain `$var` or `${var}`. This code says `{$var}`, which always leave the curly brackets there.

File: BrainPortal/cbrain_plugins/cbrain-plugins-base/boutiques_descriptors/singularity_env_demo.json (hunk 1 above)
Comment (MontrealSergiy) [reply BrainPortal/cbrain_plugins/cbrain-plugins-base/boutiques_descriptors/singularity_env_demo.json]: I removed that output to make tool seem more plausible

**Comment by natacha-beck** [comment]
One general question. Should we update the tool.config id a container engine exist ?

**Comment by MontrealSergiy** [comment]
> One general question, should we update the tool.config id a container engine exist ?

Not sure since most tools deployed in a singularity, but, If you like, ok

**Comment by prioux** [comment]
This pool request doesn't do at all what I wanted.

The original issue #1266 is about the singularity execution layer, not about Boutiques integration. The code that needs to be changed is in the CBRAIN framework.

I am closing this PR because CBRAIN doesn't need at all a solution at this level of the execution stack.

**Pull request closed without merging**